x-pack/plugins/data_visualizer/public/application/file_data_visualizer/components/import_view/import.ts

```diff
@@ -0,0 +1,243 @@
+/*
```

Comment:
This is a typescript rewrite of the import function from import_view.js.
The logic hasn't changed, it uses the `useState` function from import_view.js keep the process updated.